src/TArea.cpp

```diff
@@ -57,9 +57,9 @@ TArea::TArea(TMap* pMap, TRoomDB* pRDB)
 
 TArea::~TArea()
 {
-    if (mpRoomDB) {
+    if (mpRoomDB && !mpRoomDB->mBulkDeletionMode) {
         mpRoomDB->removeArea(this);
-    } else {
+    } else if (!mpRoomDB) {
         qDebug() << "ERROR: In TArea::~TArea(), instance has no mpRoomDB";
     }
```

Comment:
A nanoscopic performance increase might be to only do the `nullptr` check **once**. :grinning:
```diff
-    if (mpRoomDB) {
-        mpRoomDB->removeArea(this);
-    } else {
-        qDebug() << "ERROR: In TArea::~TArea(), instance has no mpRoomDB";
+    if (!mpRoomDB) {
+        qDebug() << "ERROR: In TArea::~TArea(), instance has no mpRoomDB";
+        return;
+    }
+    if (!mpRoomDB->mBulkDeletionMode) {
+        mpRoomDB->removeArea(this);
     }
```